CMakeLists.txt

```diff
@@ -13,6 +13,10 @@ endif()
 add_library(ArborX INTERFACE)
 target_link_libraries(ArborX INTERFACE Kokkos::kokkos)
 set_target_properties(ArborX PROPERTIES INTERFACE_COMPILE_FEATURES cxx_std_14)
+# As all executables using ArborX depend on it, depending on record_hash allows
+# updating hash each time executable is rebuilt, including when called from
+# within a subdirectory.
+add_dependencies(ArborX record_hash)
```

Comment:
I was surprised to see that in CMake one can use targets before they are defined.